cmd/ks-apiserver/app/options/options.go

```diff
@@ -117,13 +118,19 @@ func (s *ServerRunOptions) NewAPIServer(stopCh <-chan struct{}) (*apiserver.APIS
 	if s.MonitoringOptions == nil || len(s.MonitoringOptions.Endpoint) == 0 {
 		return nil, fmt.Errorf("moinitoring service address in configuration MUST not be empty, please check configmap/kubesphere-config in kubesphere-system namespace")
 	} else {
-		monitoringClient, err := prometheus.NewPrometheus(s.MonitoringOptions)
+		prometheusClient, err := prometheus.NewPrometheus(s.MonitoringOptions)
 		if err != nil {
 			return nil, fmt.Errorf("failed to connect to prometheus, please check prometheus status, error: %v", err)
 		}
-		apiServer.MonitoringClient = monitoringClient
+		apiServer.PrometheusClient = prometheusClient
 	}
 
+	metricsClient, metricsAPIAvailable, err := metricsserver.NewMetricsClient(kubernetesClient.Kubernetes(), s.KubernetesOptions)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to metrics-server, please check metrics-server status, error: %v", err)
```

Comment:
metrics server is a required deployment? This behavior is different, try to think this, what if a user upgrades KubeSphere but with no previous metric-server deployment, ks-apiserver will not start.

Reply:
So I think we'd better make metrics-server an optional deployment, ks-apiserver can start without metrics-server, but if this a metrics-server, we can do more things.

Reply:
Fixed